plugins/license-fact-providers/scancode/src/main/kotlin/ScanCodeLicenseFactProvider.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2021 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.plugins.licensefactproviders.scancode
+
+import java.io.File
+
+import org.apache.logging.log4j.kotlin.logger
+
+import org.ossreviewtoolkit.plugins.api.OrtPlugin
+import org.ossreviewtoolkit.plugins.api.PluginDescriptor
+import org.ossreviewtoolkit.plugins.licensefactproviders.api.LicenseFactProvider
+import org.ossreviewtoolkit.plugins.licensefactproviders.api.LicenseFactProviderFactory
+import org.ossreviewtoolkit.utils.common.Os
+import org.ossreviewtoolkit.utils.common.realFile
+
+private const val FALLBACK_DIR = "/opt/scancode-license-data"
+
+/** The configuration for the ScanCode license fact provider. */
+data class ScanCodeLicenseFactProviderConfig(
+    /**
+     * The directory that contains the ScanCode license texts. If not set, the provider will try to locate the ScanCode
+     * license text directory using a heuristic based on the path of the ScanCode binary.
+     */
+    val scanCodeLicenseTextDir: String?
```

Comment:
Probably not for now to keep this implementation closer to the current behavior, but would you agree that going forward we should probably change this implementation to lazy call `scancode-license-data` to expert ScanCode license data to a fixed / cache directory in favor of removing the heuristic?